subprojects/core/src/test/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactoryTest.groovy

```diff
@@ -563,8 +563,8 @@ class AnnotationProcessingTaskFactoryTest extends AbstractProjectBuilderSpec {
 
         where:
         type                                      | prop         | value                    | expected
-        TaskWithNestedBean                        | "bean.class" | [null] as Object[]       | Bean.class.getName()
-        TaskWithNestedBeanWithPrivateClass        | "bean.class" | [null, null] as Object[] | Bean2.class.getName()
+        TaskWithNestedBean                        | "bean.class" | [null] as Object[]       | Bean.class
+        TaskWithNestedBeanWithPrivateClass        | "bean.class" | [null, null] as Object[] | Bean2.class
```

Comment:
Should we have a case for iterable nested beans here, too?

Reply:
It would be nice to have some test somewhere that shows the full set of properties discovered for different nested beans.